test/AutoDiff/leakchecking.swift

```diff
@@ -22,7 +29,58 @@ LeakCheckingTests.test("BasicVarLeakChecking") {
     let x: Tracked<Float> = 1.0
     let _ = model.gradient(at: x) { m, x in m.applied(to: x) }
   }
-  expectEqual(0, _GlobalLeakCount.count, "Leak Detected.")
+  expectEqual(0, _GlobalLeakCount.count, "Leak detected.")
+}
+
+LeakCheckingTests.test("ControlFlow") {
+  testWithLeakChecking(expectedLeakCount: 0) {
```

Comment:
This `cond_tuple_var` test is not effective because it operates on `Float` instead of `Tracked<Float>`.
Fixing it requires significant changes to the `Tracked` type. I'll fix and re-add it (and more tests) in a follow-up.

The control flow leakchecking tests below should be sufficient for testing memory leaks for now.